cli-plugins/manager/hooks.go

```diff
@@ -0,0 +1,81 @@
+package manager
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/docker/cli/cli-plugins/hooks"
+	"github.com/docker/cli/cli/command"
+	"github.com/spf13/cobra"
+)
+
+// HookPluginData is the type representing the information
+// that plugins declaring support for hooks get passed when
+// being invoked following a CLI command execution.
+type HookPluginData struct {
+	RootCmd string
+	Flags   []string
+}
```

Comment:
Sry if I'm missing something, but couldn't spot any clarification about this. How can hooks access arguments that are not passed as flag? Eg `docker run -it my-app sh` -> how can the hook access `my-app` and `sh` or `docker pull nginx` how can the hook access `nginx` value? Or is this passed as `RootCmd` string?

Reply:
Ok, `TemplateReplaceArg` does the trick I guess, marking as resolved..